### PR TITLE
feature: Add bdbVersion/spareCores/withClang/withDebug/withGui args

### PR DESCRIPTION
**Pull request by cbergqvist**
Let me know if you want me to break the changes into separate PRs!

Example usage:
```console
$ nix-shell --arg withClang true --arg withDebug true --arg withGui false --arg spareCores 2 --argstr bdbVersion db48
```

### CODE REVIEW

**Comment by 0xB10C** [comment]
Please rebase on master for CI.

I think we would want to document the argument strings in a table in README.md? Or other ideas?

Maybe also a clang CI job would be good? Maybe replacing the job matrix with a few defined combinations..

**Comment by 0xB10C** [comment]
I was so free and rebased this for the CI to run on it. I'll have a closer look once CI passes.

**Comment by cbergqvist** [comment]
Did some minor updates + added content to README.md along the lines you specified. Let me know what you think.

Don't know if a Clang CI job is necessary. Is Clang not covered by Bitcoin Core CI?

File: shell.nix [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
+# Usage example:
+# $ nix-shell --arg withClang true --arg withDebug true
+{ pkgs ? import <nixpkgs> {},
+  withClang ? false,
+  withDebug ? false,
+  withGui ? false,
+}:
 #pkgs.clangStdenv.mkDerivation {
 #  name = "libcxxStdenv";
 # clang_13
```
Comment (0xB10C) [inline shell.nix:12]: nit: Could drop this

**Review by 0xB10C — APPROVED** [review]
Looks good. Played around with it a bit and it seems to work well. Thank you!

Any particular reason for having both `db48` and `db5`? I assumed `db48` would be sufficient.

File: shell.nix (hunk 1 above)
Comment (0xB10C) [reply shell.nix:12]: Done in #9

**Comment by 0xB10C** [comment]
> Don't know if a Clang CI job is necessary. Is Clang not covered by Bitcoin Core CI?

Clang is covered by the Core CI, a CI here would test that `nix-shell --arg withClang true` continues works fine. By default, were only testing GCC here and might not notice if something on the NixOS or Bitcoin Core side changes in regards to clang. Probably not super important though.

**Comment by cbergqvist** [comment]
> Any particular reason for having both `db48` and `db5`? I assumed `db48` would be sufficient.

Just another "test vector" since the build system supports `--with-incompatible-bdb`.

**Comment by cbergqvist** [comment]
> Clang is covered by the Core CI, a CI here would test that nix-shell --arg withClang true continues works fine.

I'm fine either way. Nice to have as long as whoever pays the CI bills is okay with it.

**Pull request merged**